tests/ha/check_cluster.pm

```diff
@@ -18,8 +18,17 @@ use hacluster;
 
 sub run {
     barrier_wait("FENCING_DONE_$cluster_name");
+
     ha_export_logs;
-    assert_script_run '(( $(grep -sR segfault /var/log | wc -l) == 0 ))';
+
+    if (!script_run '(( $(grep -sR segfault /var/log | wc -l) == 0 ))') {
```

Comment:
Shouldn't we fail here, so people will take a look into the logs and not just ignoring segfaults? Erroneous scenarios can be handled by post_fail_hook if you still want to continue with execution.

Reply:
I don't want to completely failed here, because all parts of the HA tests have been done correctly before (or we will not execute this code). This is why I changed this to record the failure in logs as a soft-failure (as well as for HA cluster health check).

But for other errors I want to stop execution, this is what is done in my post_fail_hook.

Reply:
So would it be better to move that part to its own test module and let just this test module fail but non-fatal?

Reply:
It's already in his own test module - check_cluster.pm only checks the logs and cluster health at the end - but I can try to add a specific `post_fail_hook` for this module.

I have the same `post_fail_hook` for all my HA tests (it's easier to only maintain one), but I can make an exception :)

But I would like to be able to do all tests in that module, is `assert_script_run` will not prevent the execution of my other tests?

Reply:
It's more `test_flags` that I need to change instead of `post_fail_hook`. I will testing this and update my PR.

Reply:
I tried with `test_flags` modifications, it continues beside the failing test (that's what I want), but it appears "green" and that's not really good for me, I would have prefer a soft-failure:
http://moon.qa.suse.cz/tests/2281
http://moon.qa.suse.cz/tests/overview?build=368.3&distri=sle&version=15&groupid=5

But I can see the name of the failing module beside the "green" status, and it's not the case on a good test.

So, I changed my PR with these modifications.